core/src/consensus.rs

```diff
@@ -860,30 +863,24 @@ impl Tower {
             )
         );
 
-        // return immediately if votes are empty...
-        if self.lockouts.votes.is_empty() {
-            return Ok(self);
```

Comment:
I didn't like my previous coding (especially regarding the branched-off early bail-out code flow with `Ok(_)`)...  Well, I found out that I can steamline these a bit.